lib/plugins/aws/invokeLocal/index.js

```diff
@@ -61,39 +72,38 @@ class AwsInvokeLocal {
       this.options.functionObj = this.serverless.service.getFunction(this.options.function);
       this.options.data = this.options.data || '';
 
-      return new BbPromise(resolve => {
-        if (this.options.contextPath) {
-          this.validateFile(this.options.contextPath, 'context');
-        }
-
-        if (this.options.data) {
-          return resolve();
-        } else if (this.options.path) {
-          this.validateFile(this.options.path, 'data');
-          return resolve();
-        }
+      const validatePromise =
+        (this.options.contextPath && this.validateFile(this.options.contextPath, 'context')) ||
+        BbPromise.resolve();
 
-        return stdin()
-          .then(input => {
-            this.options.data = input;
-            return resolve();
-          })
-          .catch(() => resolve());
-      }).then(() => {
-        try {
-          // unless asked to preserve raw input, attempt to parse any provided objects
-          if (!this.options.raw) {
-            if (this.options.data) {
-              this.options.data = JSON.parse(this.options.data);
-            }
-            if (this.options.context) {
-              this.options.context = JSON.parse(this.options.context);
+      return validatePromise
+        .then(() => {
+          if (this.options.data) {
+            return null;
+          } else if (this.options.path) {
+            return this.validateFile(this.options.path, 'data');
+          }
+          return stdin()
+            .then(input => {
+              this.options.data = input;
+            })
+            .catch(() => {});
```

Comment:
Let's move this callback to be second callback for previous `then` (to reflect if/else logic)